src/reaction_network.jl

```diff
@@ -531,8 +537,26 @@ function get_pexpr(parameters_extracted, options)
     pexprs = (haskey(options, :parameters) ? options[:parameters] :
               (isempty(parameters_extracted) ? :() : :(@parameters)))
     foreach(p -> push!(pexprs.args, p), parameters_extracted)
+    append!(pexprs.args, get_noise_scaling_pexpr(options))
     pexprs
 end
+# Extracts any decalred noise scaling parameters.
+function get_noise_scaling_pexpr(options)
+    haskey(options, :noise_scaling_parameters) || return []
+    ns_expr = options[:noise_scaling_parameters]
+    for idx = length(ns_expr.args):-1:3
+        if (ns_expr.args[idx] isa Symbol) || # Parameter on form η.
+           (ns_expr.args[idx] isa Expr) && (ns_expr.args[idx].head == :ref) || # Parameter on form η[1:3].
+           (ns_expr.args[idx] isa Expr) && (ns_expr.args[idx].head == :(=)) # Parameter on form η=0.1.
+            if idx < length(ns_expr.args) && (ns_expr.args[idx+1] isa Expr)  && (ns_expr.args[idx+1].head == :vect)
+                push!(ns_expr.args[idx+1].args,:(noisescalingparameter=true))
+            else
+                insert!(ns_expr.args, idx+1, :([noisescalingparameter=true]))
+            end
+        end
+    end
+    return ns_expr.args[3:end]
+end
```

Comment:
Will this work if the user gives additional metadata supported by MTK? Why is all this special code being used? Can't we do something like we do for `@species` and just create it as a normal parameter, and then add one extra line of generated code to attach the metadata?

Reply:
There are 3 cases for detecting parameters on these forms:
```
@noise_scaling_parameters p1 p2[1:2] p3=1.0
```

Yes, additional metadata is supported, see tests. E.g
```
@noise_scaling_parameters p1 [description="A parameter]
```
works.

Doing like for species might be a good idea, yeah, will check and try to copy that.

Reply:
I think I could down the species path. However, in that case it seems I might have to create a separate `noise_ps_expr` and add at the end as:
```julia
# Returns the rephrased expression.
quote
    $ps
    $noise_ps_expr
    $ivexpr
    $vars
    $sps

    Catalyst.make_ReactionSystem_internal($rxexprs, $tiv, union($spssym, $varssym),
                                          $pssym; name = $name,
                                          spatial_ivs = $sivs)
end
```

Reply:
I think that’s fine to add a line for it after the `ps`.

Reply:
Reusing the existing MTK stuff and just adding the needed metadata as an extra step seems much more future proof to me.

Reply:
Agreed, will modify the code.